cpp/src/column/column.cpp

```diff
@@ -212,13 +211,17 @@ namespace{
 }
 
 // returns the size in bytes of the specified gdf_dtype
-gdf_size_type gdf_dtype_size(gdf_dtype dtype) {
-  return cudf::type_dispatcher(dtype, get_type_size{});
+gdf_size_type gdf_dtype_size(gdf_dtype dtype)
```

Comment:
Shouldn't this function be excised in favor of `cudf::size_of`?

Reply:
Yes it should. Although - there are places in which it's used like so: ``` gdf_dtype_size(my_gdf_column_.dtype)``` - and this I'm going to replace with `cudf::byte_width(my_gdf_column_)```.

Reply:
Why can't we excise API functions? We do that all the time? libcudf is not at 1.0 yet so we can change the API at will.

Reply:
@harrism : I suppose can, but why should it be in this PR, which is about adding utility functions? Perhaps someone might be using this function from Python, or have other objections?

Reply:
All it takes is a quick grep to see that this function is exposed in the Cython bindings but never called from the Python API: 
```
(cudf_dev) mharris@dgx14:~/RAPIDS/cudf$ grep -r gdf_dtype_size python/cudf/*                                                      python/cudf/bindings/cudf_cpp.pxd:    cdef gdf_size_type gdf_dtype_size(gdf_dtype dtype) except +
```

Reply:
@harrism : I had the (apparently mistaken) notion that if something was exposed through Cython then it was already part of the API and applications / cudf-using libraries might already be using it.

Would you guys like me to remove the two API functions?

Reply:
What two? I thought we were just talking about `gdf_dtype_size`. But yes, we should remove them -- that's the only way to prevent people using them. :)

Reply:
Done.